examples/specs/layer_bar_annotations.vl.json

```diff
@@ -0,0 +1,60 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+  "description": "A simple bar chart with annotation.",
+  "data": {
+    "values": [
+      {"Day": "D01", "Value": "50"},
+      {"Day": "D02", "Value": "110"},
+      {"Day": "D03", "Value": "70"},
+      {"Day": "D04", "Value": "50"},
+      {"Day": "D05", "Value": "80"},
+      {"Day": "D06", "Value": "150"},
+      {"Day": "D07", "Value": "140"},
+      {"Day": "D08", "Value": "130"},
+      {"Day": "D09", "Value": "390"},
+      {"Day": "D10", "Value": "200"},
+      {"Day": "D11", "Value": "70"},
+      {"Day": "D12", "Value": "50"},
+      {"Day": "D13", "Value": "100"},
+      {"Day": "D14", "Value": "150"},
+      {"Day": "D15", "Value": "130"},
+      {"Threshold": "hazardous", "ThresholdValue": "300"}
+    ]
+  },
+  "transform": [{
+    "calculate": "datum.Value > 300 ? 300 : datum.Value", "as": "diff"
+  }],
+  "layer": [
+    {
+      "mark": "rule",
+      "encoding": {
+        "y": {"field": "ThresholdValue", "type": "quantitative"}
+      }
+    }, {
+      "mark": "bar",
+      "encoding": {
+        "x": {"field": "Day", "type": "ordinal"},
+        "y": {"field": "Value", "type": "quantitative"}
+      }
+    }, {
+      "mark": {
+        "type": "text",
+        "align": "left",
```

Comment:
It should be align right and dx = plot width?

It's probably worth implementing https://github.com/vega/vega-lite/issues/3022 too

Reply:
(We can just implement #3022 later, and update this example to be nicer.)

Reply:
How can I get the plot width?

Reply:
Just stick the number for now (in this PR).

But I think you should also implement #3022 so you get a cleaner syntax for width.